### PR TITLE
Create Network Security Url Lists

### PR DESCRIPTION
**Pull request by diogoEsteves42**
Adding network security urlLists resource support.

fixes: hashicorp/terraform-provider-google#14161

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_network_security_url_lists`
```

### CODE REVIEW

File: mmv1/products/networksecurity/UrlLists.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+base_url: "projects/{{project}}/locations/{{location}}/urlLists"
+create_url: "projects/{{project}}/locations/{{location}}/urlLists?urlListId={{name}}"
+self_link: "projects/{{project}}/locations/{{location}}/urlLists/{{name}}"
+min_version: beta
+update_verb: :PATCH
+update_mask: true
+description: |
+  UrlList proto helps users to set reusable, independently manageable lists of hosts, host patterns, URLs, URL patterns.
+references:
+  !ruby/object:Api::Resource::ReferenceLinks # TODO(diogoesteves): change the url to beta once it is available.
```
Comment (zli82016) [inline mmv1/products/networksecurity/UrlLists.yaml:26]: Can you please add the guide https://cloud.google.com/secure-web-proxy/docs/use-url-list? Thanks.

File: mmv1/third_party/terraform/tests/resource_network_security_url_lists_test.go.erb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+			},
+		},
+	})
+}
+
+func testAccNetworkSecurityUrlLists_basic(urlListsName string) string {
+    return fmt.Sprintf(`
+resource "google_network_security_url_lists" "foobar" {
+    name        = "%s"
+	location    = "us-central1"
```
Comment (zli82016) [inline mmv1/third_party/terraform/tests/resource_network_security_url_lists_test.go.erb:46]: The indentation is weird here.

File: mmv1/third_party/terraform/tests/resource_network_security_url_lists_test.go.erb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    values = ["www.example.com"]
+}
+`, urlListsName)
+}
+
+func testAccNetworkSecurityUrlLists_update(urlListsName string) string {
+    return fmt.Sprintf(`
+resource "google_network_security_url_lists" "foobar" {
+    name        = "%s"
+	location    = "us-central1"
```
Comment (zli82016) [inline mmv1/third_party/terraform/tests/resource_network_security_url_lists_test.go.erb:56]: The indentation is weird here.

**Comment by zli82016** [comment]
Can you please rebase the main branch to run the checks? Thanks.

File: mmv1/products/networksecurity/UrlLists.yaml (hunk 1 above)
Comment (diogoEsteves42) [reply mmv1/products/networksecurity/UrlLists.yaml:26]: Hi, added the field for guides and used the one you pointed.

File: mmv1/products/networksecurity/UrlLists.yaml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    primary_resource_id: "default"
+    vars:
+      resource_name: "my-url-lists"
+  - !ruby/object:Provider::Terraform::Examples
+    min_version: beta
+    name: "network_security_url_lists_advanced"
+    primary_resource_id: "default"
+    vars:
+      resource_name: "my-url-lists"
+parameters:
```
Comment (zli82016) [inline mmv1/products/networksecurity/UrlLists.yaml:65]: If name or location changes, it should create a new resource, right? If that is true, `immutable: true` should be added for those two properties.

Comment (diogoEsteves42) [reply mmv1/products/networksecurity/UrlLists.yaml:65]: Oh great, you are correct! Added the immutable for name and location.

**Comment by zli82016** [comment]
Hello, @diogoEsteves42 , sorry, can you please rebase the main branch and resolve the conflicts again? Thanks.

**Comment by zli82016** [comment]
Hello, @diogoEsteves42 , do you plan to reopen the PR?

**Comment by diogoEsteves42** [comment]
Yes, I just click wrong in the discard changes in the github! Taking a look on how to restore.

**Review by zli82016 — APPROVED** [review]
LGTM

**Comment by diogoEsteves42** [comment]
Thanks for the attention here @zli82016!

**Pull request merged**